hbase-server/src/test/java/org/apache/hadoop/hbase/mob/TestMobCompactionBase.java

```diff
@@ -0,0 +1,242 @@
+  * 10. Runs Mob cleaner chore
+  * 11 Verifies that number of MOB files in a mob directory is 20.
+  * 12 Runs scanner and checks all 3 * 1000 rows.
+ */
+@SuppressWarnings("deprecation")
+public abstract class TestMobCompactionBase {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(TestMobCompactionBase.class);
+
+  protected HBaseTestingUtility HTU;
+
+  protected final static String famStr = "f1";
+  protected final static byte[] fam = Bytes.toBytes(famStr);
+  protected final static byte[] qualifier = Bytes.toBytes("q1");
+  protected final static long mobLen = 10;
+  protected final static byte[] mobVal = Bytes
+      .toBytes("01234567890123456789012345678901234567890123456789012345678901234567890123456789");
+
+  protected Configuration conf;
+  protected HTableDescriptor hdt;
+  private HColumnDescriptor hcd;
+  protected Admin admin;
+  protected Table table = null;
+  protected long minAgeToArchive = 10000;
+  protected int numRegions = 20;
+  protected int rows = 1000;
+
+  protected MobFileCleanerChore cleanerChore;
+
+  public TestMobCompactionBase() {
+  }
+
+
+  @Before
+  public void setUp() throws Exception {
+    HTU = new HBaseTestingUtility();
+    hdt = HTU.createTableDescriptor(getClass().getName());
+    conf = HTU.getConfiguration();
+
+    initConf();
+
+    HTU.startMiniCluster();
+    admin = HTU.getAdmin();
+    cleanerChore = new MobFileCleanerChore();
+    hcd = new HColumnDescriptor(fam);
+    hcd.setMobEnabled(true);
+    hcd.setMobThreshold(mobLen);
+    hcd.setMaxVersions(1);
+    hdt.addFamily(hcd);
+    byte[][] splitKeys = generateSplitKeys();
+    table = HTU.createTable(hdt, splitKeys);
+
+  }
+
+  private byte[][] generateSplitKeys() {
+    RegionSplitter.UniformSplit splitAlgo = new RegionSplitter.UniformSplit();
+    return splitAlgo.split(numRegions);
+  }
+
+
+  protected void initConf() {
+
+    conf.setInt("hfile.format.version", 3);
+    // Disable automatic MOB compaction
+    conf.setLong(MobConstants.MOB_COMPACTION_CHORE_PERIOD, 0);
+    // Disable automatic MOB file cleaner chore
+    conf.setLong(MobConstants.MOB_CLEANER_PERIOD, 0);
+    // Set minimum age to archive to 10 sec
+    conf.setLong(MobConstants.MIN_AGE_TO_ARCHIVE_KEY, minAgeToArchive);
+    // Set compacted file discharger interval to a half minAgeToArchive
+    conf.setLong("hbase.hfile.compaction.discharger.interval", minAgeToArchive/2);
+  }
+
+  private void loadData(int num) {
+
+    Random r = new Random();
+    try {
+      LOG.info("Started loading {} rows", num);
+      for (int i = 0; i < num; i++) {
+        byte[] key = new byte[32];
+        r.nextBytes(key);
+        Put p = new Put(key);
+        p.addColumn(fam, qualifier, mobVal);
+        table.put(p);
+      }
+      admin.flush(table.getName());
+      LOG.info("Finished loading {} rows", num);
+    } catch (Exception e) {
+      LOG.error("MOB file compaction chore test FAILED", e);
+      assertTrue(false);
```

Comment:
use `fail` and Include text about why we're failing, so that junit can propagate the message.